lib/Duo/API/Iterator.pm

```diff
@@ -62,23 +62,23 @@ Returns: a list of all remaining items not yet returned by the generator.
 
 =cut
 
-use Moo;
+use warnings;
+use strict;
 use Carp qw(croak);
-use Scalar::Util qw(reftype);
-
-has 'generator' => (
-    is => 'ro',
-    isa => sub {
-        my $input = shift;
-        croak "The generator parameter must be a subroutine references."
-            unless reftype($input) eq 'CODE' 
-    },
-    required => 1
-);
+
+sub new {
+    my ($class,%params) = @_;
+    my $g = 'generator';
+    croak "Missing required arguments: $g"
+        unless exists $params{$g};
+    croak "The $g parameter must be a subroutine reference."
+        unless ref($params{$g}) eq 'CODE';
```

Comment:
Unless there's a compelling reason I'd rather leave this as Scalar::Util since this will reject blessed subroutines.